Reduce header includes in Document.h

rniwa opened the pull request:
#### f3cbd6853a8f3a93bec297b808161a1bace3fa76
<pre>
Reduce header includes in Document.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=292426">https://bugs.webkit.org/show_bug.cgi?id=292426</a>

Reviewed by Mark Lam.

Remove more #include from Document.h to speed up build time.

* Source/WebCore/Modules/WebGPU/GPUQueue.cpp:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::fragmentDirectiveForBindings):
* Source/WebCore/dom/Document.h:
(WebCore::Document::fragmentDirectiveForBindings): Moved to cpp.
(WebCore::Document::syncData): Deleted.
* Source/WebCore/dom/DocumentInlines.h:
(WebCore::Document::syncData):
* Source/WebCore/page/IntersectionObserver.h:
* Source/WebCore/page/Page.h:
* Source/WebCore/platform/graphics/ca/cocoa/GraphicsLayerAsyncContentsDisplayDelegateCocoa.h:
* Source/WebCore/rendering/BackgroundPainter.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebProcessSyncClient.cpp:

Canonical link: <a href="https://commits.webkit.org/294465@main">https://commits.webkit.org/294465@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cd8119f5391fe2f7b36c1b31bc7c6b481b5572eb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101872 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21540 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11856 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/107030 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52506 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/103912 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21848 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30047 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77565 "Found 1 new test failure: imported/w3c/web-platform-tests/html/semantics/forms/constraints/infinite_backtracking.html (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34579 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104879 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16882 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91991 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57903 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16708 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10014 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51858 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86560 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10089 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109397 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/29005 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21363 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/86550 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29366 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88195 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86127 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30872 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8597 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/23176 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16575 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28933 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34228 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28744 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32067 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30303 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

MenloDorian commented on Source/WebCore/dom/Document.h:
```diff
@@ -34,16 +34,11 @@
 #include "FrameDestructionObserver.h"
 #include "FrameIdentifier.h"
 #include "HitTestSource.h"
-#include "IntDegrees.h"
```
Should this be included in Document.cpp to make sure the `using IntDegrees = int32_t;` never goes out of alignment with what's defined in the header?

rniwa replied:
Sure, we can do that.

MenloDorian reviewed (approved):
r=me with a suggestion for improvement.

rniwa committed:
Reduce header includes in Document.h
https://bugs.webkit.org/show_bug.cgi?id=292426

Reviewed by Mark Lam.

Remove more #include from Document.h to speed up build time.

* Source/WebCore/Modules/WebGPU/GPUQueue.cpp:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::fragmentDirectiveForBindings):
* Source/WebCore/dom/Document.h:
(WebCore::Document::fragmentDirectiveForBindings): Moved to cpp.
(WebCore::Document::syncData): Deleted.
* Source/WebCore/dom/DocumentInlines.h:
(WebCore::Document::syncData):
* Source/WebCore/page/IntersectionObserver.h:
* Source/WebCore/page/Page.h:
* Source/WebCore/platform/graphics/ca/cocoa/GraphicsLayerAsyncContentsDisplayDelegateCocoa.h:
* Source/WebCore/rendering/BackgroundPainter.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebProcessSyncClient.cpp:

Canonical link: https://commits.webkit.org/294465@main
